### PR TITLE
Set up stack project

### PR DESCRIPTION
**Pull request by C0Florent**
This PR properly sets up stack by declaring the used stack version in `stack.yaml`

It also declares a reproducible development environment with nix (flake.nix and flake.lock files).

---

*Note*: Should be merged **after** #1

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,7 +5,24 @@
...
+
+fclean:
+	stack purge
+
+re: fclean all
+
+tests_run:
+	stack test
+
+.PHONY: $(TARGET) clean fclean re tests_run
```
Comment (ThibautHH) [inline Makefile:28]: `all` should be in `.PHONY` too.
Maybe consider investigating if it's possible to get the used source files from a `stack` command to make them a proper dependency of the executable target ?
Though I think I remember Stack avoiding unecessary re-builds of files by itself anyway.

File: Makefile
Comment (ThibautHH) [inline Makefile:1]: I'd say the recipes' commands should be silenced with `@` (or the `.SLIENT` target)

**Review by ThibautHH — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (C0Florent) [reply Makefile:28]: Not putting `all` in `.PHONY` is an oversight, thanks for pointing it out

I think it's not worth the hassle of tracking dependencies of targets, let's keep this Makefile a simple wrapper around stack

File: Makefile
Comment (C0Florent) [reply Makefile:1]: I'm ok with that, I'll make the change

File: Makefile (hunk 1 above)
Comment (ThibautHH) [reply Makefile]: Sure.

**Review by ThibautHH — APPROVED** [review]

**Pull request merged**